### ISSUE TITLE
closeOnSelect is not working with 2.8.4

### ISSUE THREAD

**Issue by vijayst** [YouCanBookMe/react-datetime#263]
closeOnSelect is not working after upgrading to 2.8.4.

**Comment by simeg** [comment]
Hi @vijayst. Exactly what props are you using? Because [in this demo](http://codepen.io/simeg/pen/wgEVaP) it works as intended.

**Comment by mdkwock** [comment]
I think it's related to the fix for https://github.com/YouCanBookMe/react-datetime/issues/163

from what I'm seeing, it's possible to hit a race condition between the setState({open: false}) from the closeCalendar function and the componentWillReceiveProps resetting the open state to be true.

what is the reason for resetting the open state to the existing state, rather than just not setting the open state if it's not defined?

**Comment by vijayst** [comment]
@simeg The props that I am passing are:
```
<Datetime
	viewMode={this.props.viewMode}
	dateFormat={this.props.dateFormat}
	timeFormat={this.props.timeFormat}
	closeOnSelect={true}
	value={this.state.value ? moment(this.state.value).format(this.props.dateFormat) : ''}
	onChange={this.onChange.bind(this)}
	valueFormat={this.props.valueFormat}
	inputProps={{
		...this.props.inputProps,
		className: this.props.small ? 'form-control input-sm' : 'form-control',
		readOnly: true,
		disabled: this.props.disabled,
		ref: (c) => { this.inputComponent = c }
	}}
/>
```

**Comment by pnaveen10** [comment]
I'm also seeing the same issue after upgrading to the recent version.

The props I'm using are:
`<Datetime
	timeFormat={false}
	closeOnSelect={true}
	value={this.state.date}
	onChange={this.updateDate}
	inputProps={{ placeholder: 'mm/dd/yyyy'}}
/>`

**Comment by AndreyShpilevoy** [comment]
Also faced with the same problem.
If I delete `onChange={ onChange }` it's begun to work again.

**Comment by simeg** [comment]
Thanks for the information everyone. I can't seem to reproduce this, if you could help me we could squash this bug a lot faster. I've tried all the props you guys have written here but no go. If someone could fork this codepen and show me the bug I'd be very grateful. Please use 2.8.3 during the meantime, sorry for the inconvenience.

http://codepen.io/simeg/pen/wgEVaP

**Comment by mdkwock** [comment]
there might be a simpler example but this one reproduces the behavior I see
https://codepen.io/anon/pen/QdZJvG

**Comment by simeg** [comment]
@mdkwock thanks for that.

I published a fix for this problem in 2.8.5. The logic can sometimes be hard to follow in this codebase, I think. Therefore it's hard to foresee all cases.

I'll keep this issue open until at least one or two of you confirms that the new version works.

**Comment by simeg** [comment]
Hey again, my fix was not a good one so I reverted some commits that brings us back to the logic we had a while back. So this specific issue should hopefully be resolved in 2.8.6. Sorry for any inconvenience.

**Comment by simeg** [comment]
Can anyone confirm that this is working again?

**Comment by pnaveen10** [comment]
This works fine for me. Thanks for the fix!
